VimeoUpload/Descriptor System/DescriptorManager.swift

```diff
@@ -148,14 +162,31 @@ open class DescriptorManager: NSObject
                 {
                     return
                 }
-
-                strongSelf.archiver.removeAll()
                 
                 // TODO: Need to respond to this notification [AH] 2/22/2016 (remove from downloads store, delete active uploads etc.)
 
-                strongSelf.delegate?.sessionDidBecomeInvalid?(error: error as NSError)
+                // Why do we need to check if `error` is `nil` even though the compiler
+                // tells us that this checking will always succeed? Behind the scene, we
+                // are using the `AFURLSessionManager` class -- which is an Objective-C
+                // class -- for managing background upload sessions. For the session
+                // invalid callback, its header file does not mark the `NSError` object
+                // as nullable; in reality, this object will be `nil` if we explicitly
+                // invalidate the underlying session. Because of that, it is necessary
+                // to check for `nil` here, else the runtime will crash if the error
+                // object is `nil`. [VN] (06/13/2018) 
+                let theError: NSError?
```

Comment:
I wonder if there's another way we can work around this. It's a bummer to have to intentionally introduce a warning.

Reply:
Unfortunately there might not be another solution, unless we fork out `AFNetworking` and edit that header file ourselves.